src/data/complex/basic.lean

```diff
@@ -288,6 +288,9 @@ def of_real : ℝ →+* ℂ := ⟨coe, of_real_one, of_real_mul, of_real_zero, o
 
 @[simp] lemma of_real_eq_coe (r : ℝ) : of_real r = r := rfl
 
+theorem of_real_injective : function.injective (of_real : ℝ → ℂ) :=
```

Comment:
```suggestion
theorem of_real_injective : function.injective (coe : ℝ → ℂ) :=
```
since that's the simp-normal form.

Also, it might be easier to prove `of_real_inj` in terms of this result instead.

Reply:
hmm, but this doesn't unify well with things that use the coe of some hom (like [here](https://github.com/leanprover-community/mathlib/blob/5e2bd2f71953056e5c22f89c762cd39fec7fa485/src/temp.lean#L96)). is it worth having both?

Reply:
Maybe, but you could always just use `(show function.injective of_real, from of_real_injective)` in that particular place.
I think generally we don't restate injectivity about bundled versions of maps.

Reply:
`of_real_inj` is a super easy proof, so I left it untouched.